### PR TITLE
Add mattermost service support

### PR DESCRIPTION
**Pull request by matejzero**
Add support for mattermost service.

I hope this is all I need to do:)

### CODE REVIEW

File: gitlab/v3/objects.py [hunk 1]
```diff
@@ -1675,6 +1675,7 @@ class ProjectService(GitlabObject):
 
                  # Optional fields
                  'username', 'password', 'jira_issue_transition_id')),
+        'mattermost': (('webhook',), ('username', 'channel')),
```
Comment (gpocentek) [inline gitlab/v3/objects.py]: I can't find references to `username` and `channel` in the gitlab documentation and code. Is this really supported?

Comment (matejzero) [reply gitlab/v3/objects.py]: Looking at [services API](https://docs.gitlab.com/ee/api/services.html#mattermost-notifications), it looks like `username` and `channel` are optional fields.

Comment (gpocentek) [reply gitlab/v3/objects.py]: You're right. I guess I missed it.

**Comment by matejzero** [comment]
It seems like this is not working, but the problem is in GitLab. At the moment, even if we send username / channel parameters to GitLab, they are not set. Strange, since it's documented in docs.

I opened an issue on [gitlab](https://gitlab.com/gitlab-org/gitlab-ce/issues/39895).

**Comment by matejzero** [comment]
Issue is fixed in GitLab.

**Comment by gpocentek** [comment]
Thanks for the feedback @matejzero

**Pull request merged**